### PR TITLE
Sign/Verify message for DAO bonded roles

### PR DESCRIPTION
**Pull request by ghost**
Adds the ability to sign and verify a message for DAO bonded roles.  Required for https://github.com/bisq-network/bisq2/issues/68
Reuses the same code for sign/verify DAO proof of burn txs.

image

image

image

### CODE REVIEW

File: desktop/src/main/java/bisq/desktop/main/dao/bonding/roles/RolesListItem.java [hunk 1]
```diff
@@ -75,4 +75,12 @@ public boolean isLockupButtonVisible() {
     public boolean isRevokeButtonVisible() {
         return this.daoFacade.isMyRole(this.getRole()) && (this.bondedRole.getBondState() == BondState.LOCKUP_TX_CONFIRMED);
     }
+
+    public boolean isSignButtonVisible() {
+        return this.daoFacade.isMyRole(this.getRole()) && (this.bondedRole.getBondState() == BondState.LOCKUP_TX_CONFIRMED);
+    }
+
+    public boolean isVerifyButtonVisible() {
+        return getLockupTxId() != null && getLockupTxId().length() > 1;
```
Comment (chimp1984) [inline desktop/src/main/java/bisq/desktop/main/dao/bonding/roles/RolesListItem.java]: Shouldn't be here the same as in isSignButtonVisible and isRevokeButtonVisible ?
Could be extracted to one method.

Comment (chimp1984) [reply desktop/src/main/java/bisq/desktop/main/dao/bonding/roles/RolesListItem.java]: Ah the unlocked case is different to isRevokeButtonVisible...
Maybe to make it more explicit to add methods for the possible states?
- not locked up
- locked up
- unlocked

**Review by chimp1984 — CHANGES_REQUESTED** [review]
Looks good to me. Small nits...

**Review by ripcurlx — APPROVED** [review]
utACK

**Pull request merged**